pypeit/flatfield.py

```diff
@@ -203,10 +295,32 @@ def show(self, slits=None, wcs_match=True):
                 msgs.warn('Could not load slits to show with flat-field images. Did you provide the master info??')
         if slits is not None:
             slits.mask_flats(self)
-            illumflat = self.fit2illumflat(slits)
-        # Show
-        show_flats(self.pixelflat, illumflat, self.procflat, self.flat_model,
-                   wcs_match=wcs_match, slits=slits)
+            illumflat_pixel = self.fit2illumflat(slits, frametype='pixel')
+            if self.illumflat_spat_bsplines is not None:
+                illumflat_illum = self.fit2illumflat(slits, frametype='illum')
+        # Decide which frames should be displayed
+        if type == 'pixel':
+            image_list = zip([self.pixelflat_norm, illumflat_pixel, self.pixelflat_raw, self.pixelflat_model,
+                              self.pixelflat_spec_illum],
+                             ['pixelflat_norm', 'pixelflat_spat_illum', 'pixelflat_raw', 'pixelflat_model',
+                              'pixelflat_spec_illum'],
+                             [(0.9, 1.1), (0.9, 1.1), None, None,
+                              (0.8, 1.2)])
+        elif type == 'illum':
+            image_list = zip([illumflat_illum, self.illumflat_raw],
+                             ['illumflat_spat_illum', 'illumflat_raw'],
+                             [(0.9, 1.1), None])
+        else:
+            # Show everything that's available (anything that is None will not be displayed)
+            image_list = zip([self.pixelflat_norm, illumflat_pixel, self.pixelflat_raw, self.pixelflat_model,
+                              self.pixelflat_spec_illum, illumflat_illum, self.illumflat_raw],
+                             ['pixelflat_norm', 'pixelflat_spat_illum', 'pixelflat_raw', 'pixelflat_model',
+                              'pixelflat_spec_illum', 'illumflat_spat_illum', 'illumflat_raw'],
+                             [(0.9, 1.1), (0.9, 1.1), None, None,
+                              (0.8, 1.2), (0.9, 1.1), None])
+        # Display frames
+        show_flats(image_list, wcs_match=wcs_match, slits=slits)
```

Comment:
To what extent has this been tested?  GUIs are often neglected.  :)

Reply:
I've tested all combinations of this, and tested from the script. So, I think this is OK.

Reply:
Great. I would run pypeit_chk_flats on a few setups after dev suite is done. Hard to test GUIs of course.